File: backend/app/schemas/pagos.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class CrearPagoRequest(BaseModel):
    cita_id: UUID
    tipo: str  # 'deposito' | 'saldo' | 'total' | 'penalizacion' | 'reembolso'
    metodo: str  # 'efectivo' | 'transferencia' | 'yape' | 'plin' | 'tarjeta'
    monto: float = Field(gt=0)
    referencia_externa: str | None = None


class ConfirmarPagoRequest(BaseModel):
    referencia_externa: str | None = None
    nota_admin: str | None = None


class PagoResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    cita_id: UUID
    cliente_id: UUID
    tipo: str
    metodo: str
    estado: str
    monto: float
    referencia_externa: str | None
    comprobante_url: str | None
    confirmado_por: UUID | None
    fecha_confirmacion: datetime | None
    nota_admin: str | None

File: frontend/src/types/pagos.ts
export type TipoPago = "deposito" | "saldo" | "total" | "penalizacion" | "reembolso";
export type MetodoPago = "efectivo" | "transferencia" | "yape" | "plin" | "tarjeta";
export type EstadoPago = "pendiente" | "confirmado" | "rechazado" | "reembolsado";

export interface IPago {
  id: string;
  cita_id: string;
  cliente_id: string;
  tipo: TipoPago;
  metodo: MetodoPago;
  estado: EstadoPago;
  monto: number;
  referencia_externa: string | null;
  comprobante_url: string | null;
  confirmado_por: string | null;
  fecha_confirmacion: string | null;
  nota_admin: string | null;
}

export interface IPagoCreate {
  cita_id: string;
  tipo: TipoPago;
  metodo: MetodoPago;
  monto: number;
  referencia_externa?: string;
}

export interface IConfirmarPago {
  referencia_externa?: string;
  nota_admin?: string;
}

export interface IFiltrosPagos {
  cita_id?: string;
  estado?: EstadoPago;
}
